src/commonMain/kotlin/org/hisp/dhis/lib/expression/ExpressionMode.kt

```diff
@@ -38,7 +38,10 @@ enum class ExpressionMode(
         ValueType.BOOLEAN,
         ValueType.STRING,
         ValueType.NUMBER,
-        ValueType.DATE);
+        ValueType.DATE),
+
+    // android custom intent request parameters
+    ANDROID_CUSTOM_INTENT_EXPRESSION(ExpressionGrammar.AndroidCustomIntentMode, ValueType.NUMBER);
```

Comment:
When we discussed this you said you pass along the value typed using different types. In that case all the types you use should be listed here. At least STRING, NUMBER and BOOLEAN seem likely. 

If I recall correctly this list is for validation so we can check that the result we got matches the expectation. If you don't list a type here but the expression does compute to a value of that type and you validate you will get an error.